File: Year 2/Semester 2/MPP/Protobuf/ProtoCSharp/AirlineServer/Services/IService.cs
using Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    public interface IService
    {
        Employee login(Employee employee, IObserver client);
        void logout(Employee employee);
        List<Flight> getFlights(string destination, string date);
        Employee getEmployee(string username, string password);
        void buyTicket(int flightId, string tourist_name, string client_address);
        int getNumberOfTicketsForFlights(int flightId);
        List<Flight> getAllFlights();
    }
}

File: Year 2/Semester 2/MPP/RestServices/Java REST/AirlineServices/src/main/java/airline/services/IService.java
package airline.services;


import airline.model.Employee;
import airline.model.Flight;

public interface IService {

    Employee login(Employee employee, IAirlineObserver client) throws Exception;
    void logout(Employee employee, IAirlineObserver client) throws Exception;
    Iterable<Flight> getFlights(String destination, String date) throws Exception;
    Employee getEmployee(String username, String password);
    void buyTicket(int flightId,String tourist_name, String client_address) throws Exception;
    int getNumberOfTicketsForFlight(Integer flightId) throws Exception;
    Iterable<Flight> getAllFlights() throws Exception;
}
